tests/rptest/tests/datalake/schemas/data_types.py

```diff
@@ -0,0 +1,336 @@
+    """
+    A generic int type.
+    """
+    @staticmethod
+    def name() -> str:
+        return "generic_int"
+
+    @staticmethod
+    def to_avro() -> str:
+        return "int"
+
+    @staticmethod
+    def to_proto() -> str:
+        return "int32"
+
+    @staticmethod
+    def random(mn, mx) -> int:
+        return random.randint(mn, mx)
+
+
+class GenericLong(GenericDataType):
+    """
+    A generic long type.
+    """
+    @staticmethod
+    def name() -> str:
+        return "generic_long"
+
+    @staticmethod
+    def to_avro() -> str:
+        return "long"
+
+    @staticmethod
+    def to_proto() -> str:
+        return "int64"
+
+    @staticmethod
+    def random(mn, mx) -> int:
+        return random.randint(mn, mx)
+
+
+class GenericFloat(GenericDataType):
+    """
+    A generic float type.
+    """
+    @staticmethod
+    def name() -> str:
+        return "generic_float"
+
+    @staticmethod
+    def to_avro() -> str:
+        return "float"
+
+    @staticmethod
+    def to_proto() -> str:
+        return "float"
+
+    @staticmethod
+    def random(mn, mx) -> float:
+        return random.uniform(mn, mx)
+
+
+class GenericDouble(GenericDataType):
+    """
+    A generic double type.
+    """
+    @staticmethod
+    def name() -> str:
+        return "generic_double"
+
+    @staticmethod
+    def to_avro() -> str:
+        return "double"
+
+    @staticmethod
+    def to_proto() -> str:
+        return "double"
+
+    @staticmethod
+    def random(mn, mx) -> float:
+        return random.uniform(mn, mx)
+
+
+class GenericBytes(GenericDataType):
+    """
+    A generic bytes type.
+    """
+    @staticmethod
+    def name() -> str:
+        return "generic_bytes"
+
+    @staticmethod
+    def to_avro() -> str:
+        return "bytes"
+
+    @staticmethod
+    def to_proto() -> str:
+        return "bytes"
+
+    @staticmethod
+    def random(n=10) -> str:
+        return ''.join(
+            random.choices(string.ascii_lowercase + string.ascii_uppercase +
+                           string.digits,
+                           k=n))
+
+
+class GenericString(GenericDataType):
+    """
+    A generic string type.
+    """
+    @staticmethod
+    def name() -> str:
+        return "generic_string"
+
+    @staticmethod
+    def to_avro() -> str:
+        return "string"
+
+    @staticmethod
+    def to_proto() -> str:
+        return "string"
+
+    @staticmethod
+    def random(n=10) -> str:
+        return ''.join(
+            random.choices(string.ascii_lowercase + string.ascii_uppercase +
+                           string.digits,
+                           k=n))
+
+
+ALL_PRIMITIVE_DATA_TYPES = [
+    GenericBool, GenericInt, GenericLong, GenericFloat, GenericDouble,
+    GenericString
+]
+
+ALL_PRIMITIVE_DATA_TYPES_AND_ENUM = ALL_PRIMITIVE_DATA_TYPES + [GenericEnum]
+
+AVRO_PRIMITIVE_DATA_TYPES_RANDOM_GENS = {
+    gen_type.to_avro(): gen_type.random
+    for gen_type in ALL_PRIMITIVE_DATA_TYPES_AND_ENUM
+}
```

Comment:
this is cool. it's all written now so no reason to change it, but i do wonder if you could have consolidated all of this into a raw dictionary (more compact representation) and then dynamically generated the classes.